lib/ansible/parsing/quoting.py

```diff
@@ -17,15 +17,31 @@
 
 # Make coding more python3-ish
 from __future__ import (absolute_import, division, print_function)
+
+from ansible.module_utils import six
+
 __metaclass__ = type
 
 
 def is_quoted(data):
+    if not isinstance(data, six.string_types) or not isinstance(
+            data, six.binary_type):
```

Comment:
Looking more closely at the next `return len...` line, it will only work with `string_types` on python3. It won't work with `binary_type`. As such, we probably don't need to worry about checking `binary_type`

Reply:
Sorry it's fixed, thanks again for the review.